src/Common/src/Interop/OSX/Interop.EventStream.cs

```diff
@@ -142,25 +144,77 @@ internal enum FSEventStreamCreateFlags : uint
         /// <param name="runLoop">The RunLoop to attach the stream to</param>
         /// <param name="runLoopMode">The mode of the RunLoop; this should usually be kCFRunLoopDefaultMode. See the documentation for RunLoops for more info.</param>
         [DllImport(Interop.Libraries.CoreServicesLibrary)]
-        internal static extern void FSEventStreamScheduleWithRunLoop(
+        private static extern void FSEventStreamScheduleWithRunLoop(
             FSEventStreamRef    streamRef,
             CFRunLoopRef        runLoop,
             CFStringRef         runLoopMode);
 
         /// <summary>
+        /// Attaches an EventStream to a RunLoop so events can be received. This should usually be the current thread's RunLoop.
+        /// </summary>
+        /// <param name="streamRef">The stream to attach to the RunLoop</param>
+        /// <param name="runLoop">The RunLoop to attach the stream to</param>
+        /// <param name="runLoopMode">The mode of the RunLoop; this should usually be kCFRunLoopDefaultMode. See the documentation for RunLoops for more info.</param>
+        internal static void FSEventStreamScheduleWithRunLoop(
+            SafeEventStreamHandle streamRef,
+            CFRunLoopRef runLoop,
+            SafeCreateHandle runLoopMode)
+
+        {
+            FSEventStreamScheduleWithRunLoop(streamRef.DangerousGetHandle(), runLoop, runLoopMode.DangerousGetHandle());
```

Comment:
It should; you use the SafeHandle type in the signature, e.g. as a parameter type, and the runtime will handle extracting the IntPtr from it and passing that a long to the native code, doing the appropriate add ref / releasing in the process.